dnf5/commands/repoquery/repoquery.cpp

```diff
@@ -306,6 +326,46 @@ void RepoqueryCommand::set_argument_parser() {
     disable_modular_filtering = std::make_unique<libdnf5::cli::session::BoolOption>(
         *this, "disable-modular-filtering", '\0', "Include packages of inactive module streams.", false);
 
+    // Allowed values for --providers-of options (these package attributes return ReldepLists)
+    std::vector<std::string> pkg_attrs_options{
+        "conflicts",
+        "depends",
+        "enhances",
+        "obsoletes",
+        "provides",
+        "recommends",
+        "requires",
+        "requires_pre",
+        "suggests",
+        "supplements",
+        "",  // empty when option is not used
+    };
+    providers_of_option = dynamic_cast<libdnf5::OptionEnum<std::string> *>(
+        parser.add_init_value(std::make_unique<libdnf5::OptionEnum<std::string>>("", pkg_attrs_options)));
+    auto * providers_of = parser.add_new_named_arg("providersof");
+    std::string allowed_values = libdnf5::utils::string::join(pkg_attrs_options, ", ");
+    // Remove last empty space
+    allowed_values.pop_back();
+    // change last ", " to '.'
+    allowed_values.back() = '.';
+    providers_of->set_description(
+        "After filtering is finished get selected attribute of packages and output packages that provide it. "
+        "Supports: " +
+        allowed_values);
+    providers_of->set_has_value(true);
+    providers_of->set_long_name("providers-of");
+    providers_of->link_value(providers_of_option);
+    providers_of->set_arg_value_help("PACKAGE_ATTRIBUTE");
+    cmd.register_named_arg(providers_of);
+
+    recursive = std::make_unique<libdnf5::cli::session::BoolOption>(
+        *this,
+        "recursive",
+        '\0',
+        "For --whatrequires/--providers-of=requires options recursively get requirers/providers of requires of "
+        "outputed packages.",
```

Comment:
Sounds bit cumbersome to me. What about just something like: "Used with --whatrequires or --providers-of=requires options to query the packages recursively."

Reply:
Updated. :+1: